pages/curriculum/[lessonSlug]/[subLessonSlug].tsx

```diff
@@ -130,15 +172,25 @@ export const getStaticProps: GetStaticProps<any, Slugs> = async context => {
   const subLessons = (
     await Promise.all(
       slugs.map(async slug => {
+        const source = await getSubLessonContent(slug)
+
         // Only include source data on selected subLesson
         const sourceAndFrontMatter = await parseMDX(
-          await getSubLessonContent(slug),
+          source,
           slug.subLessonSlug !== subLessonSlug
         )
 
+        const { content } = matter(source)
+        const headings = content?.match(/^(#+)\s+(.*)$/gm)?.map(heading => {
+          // istanbul ignore next
```

Comment:
I need help here. Jest is complaining that this line isn't covered, but I couldn't figure out how because if the regex for `content` doesn't match any of the string and returns `null`, the regex for `heading` won't run because they're checking for almost the same pattern: if the substring contains a hash symbol with a space between it and another piece of text.

Reply:
Use a matchAll, to get the capturing groups along with the matches the first time you run the regex instead of doing it twice.

like this
```js
const headings = [...content?.matchAll(/^(#+)\s+(.*)$/gm)].map(
  ([, hashes, text]) => ({
    text,
    depth: hashes.length,
  })
);
```